addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -12,7 +12,15 @@ class ResourceCalendarLeaves(models.Model):
 
     timesheet_ids = fields.One2many('account.analytic.line', 'global_leave_id', string="Analytic Lines")
 
-    def _work_time_per_day(self):
+    def _get_resource_calendars(self):
+        leaves_with_calendar = self.filtered('calendar_id')
+        calendars = leaves_with_calendar.calendar_id
+        leaves_wo_calendar = self - leaves_with_calendar
+        if leaves_wo_calendar:
+            calendars += self.env['resource.calendar'].search([('company_id', 'in', leaves_wo_calendar.company_id.ids)])
+        return calendars
+
+    def _work_time_per_day(self, resource_calendars=False):
```

Comment:
Now that I see it again, why do we need this `resource_calendars` argument?
Each time we call `some_leave._work_time_per_day()`, we give it the result of `some_leave._get_resource_calendars()` as argument.
So for that matter, we could have no argument, and do :
`resource_calendars = self._work_time_per_day()`
inside this method. Am I overlooking something?
Sorry I didn't notice that earlier.

Reply:
Ok yes, it definitely makes sense, sorry.
And yes, it was a bad idea, my bad again :sweat_smile:

Reply:
```suggestion
    def _work_time_per_day(self, resource_calendars):
```
So please revert this and we should be good to go!